### ISSUE TITLE
Training process

### ISSUE THREAD

**Issue by Runsong123** [baegwangbin/surface_normal_uncertainty#2]
Hi Gwangbin,

Thanks for sharing your great work!
I am curious about the training process, is the network trained by end-to-end training?

Best wishes,
Runsong

**Comment by baegwangbin** [comment]
Hi Runsong,

The network generates three intermediate outputs (H/8 x W/8, H/4 x W/4, H/2 x W/2) before producing the final full-resolution output. The training loss is applied to all four predictions (hence not end-to-end). For the coarsest prediction (H/8 x W/8), the loss is applied to all pixels. For others, the loss is computed only for the pixels selected via the proposed uncertainty-guided sampling.

Regards,
Gwangbin

**Comment by Runsong123** [comment]
Got it! 
Thanks for your quick reply!  :)

Best,
Runsong
